### PR TITLE
[MOBILE-2533] Update Unity to latest SDKs

### PR DESCRIPTION
**Pull request by Apekka**
### What do these changes do?
Update to latest SDKs versions
Add preference center
Remove location
Add the Privacy manager methods

### Why are these changes necessary?
To keep the SDK updated

### How did you verify these changes?
Edited the UrbanAirship behaviour script launched at start and called all the methods, verifying that everything is run correctly

### CODE REVIEW

File: Assets/Plugins/iOS/UAUnityPlugin.m [hunk 1]
```diff
@@ -529,26 +506,71 @@ - (void)inboxUpdated {
...
 
-bool UAUnityPlugin_isPushTokenRegistrationEnabled() {
-    UA_LDEBUG(@"UnityPlugin isPushTokenRegistrationEnabled");
-    return [UAPush shared].pushTokenRegistrationEnabled ? true : false;
+#pragma mark -
+#pragma mark Preference Center
+
+void UAUnityPlugin_openPreferenceCenter(NSString *preferenceCenterId) {
+    UA_LDEBUG(@"UAUnityPlugin openPreferenceCenter");
+    //[[UAPreferenceCenter shared] openPreferenceCenter:preferenceCenterId];
```
Comment (Ulrico972) [inline Assets/Plugins/iOS/UAUnityPlugin.m:569]: I think you forgot the comment here

File: Assets/UrbanAirship/Platforms/UAirship.cs [hunk 2]
```diff
@@ -418,29 +392,52 @@ public AttributeEditor EditNamedUserAttributes ()
...
-            get {
-                return plugin.PushTokenRegistrationEnabled;
-            }
-            set {
-                plugin.PushTokenRegistrationEnabled = value;
-            }
+        /// Sets the enabled SDK features
+        /// </summary>
+        /// <param name="enabledFeatures">The features to enable</param>
+        public void SetEnabledFeatures (string[] enabledFeatures) {
```
Comment (Ulrico972) [inline Assets/UrbanAirship/Platforms/UAirship.cs:406]: Could we use an enum for the string features ?

**Review by Ulrico972 — COMMENTED** [review]
A couple of suggestions and things to change

File: Assets/UrbanAirship/Platforms/IUAirshipPlugin.cs [hunk 3]
```diff
@@ -94,6 +74,18 @@ interface IUAirshipPlugin {
...
+        void OpenPreferenceCenter (string preferenceCenterId);
+
+        void SetEnabledFeatures (string[] enabledFeatures);
+
+        void EnableFeatures (string[] enabledFeatures);
+
+        void DisableFeatures (string[] disabledFeatures);
+
+        bool IsFeatureEnabled (string feature);
+
```
Comment (jyaganeh) [inline Assets/UrbanAirship/Platforms/IUAirshipPlugin.cs:86]: Should we also have `isAnyEnabled` here?

**Review by jyaganeh — COMMENTED** [review]
looks good, overall! had one question about the `isAnyEnabled` method in the SDKs not being wrapped here and agree that `IsFeatureEnabled` should probably take a list to better match the `PrivacyManager` APIs in the SDKs.

File: Assets/Plugins/iOS/UAUnityPlugin.m [hunk 4]
```diff
@@ -25,7 +25,7 @@ + (void)load {
 
 + (void)performTakeOff:(NSNotification *)notification {
     UA_LDEBUG(@"UnityPlugin taking off");
-    [UAirship takeOff];
+    [UAirship takeOff:nil launchOptions:nil];
```
[the remark this answers is not in the document]
Comment (Ulrico972) [reply Assets/Plugins/iOS/UAUnityPlugin.m:28]: I think you should use the load method too for the take off, I think that's what we do on the other frameworks:
https://github.com/urbanairship/ios-library-dev/blob/main/Documentation/Migration/migration-guide-14-16.md#take-off

File: airship.properties [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
-version = 8.1.0
+version = 9.0.0
 
 # Urban Airship iOS SDK version
-iosAirshipVersion = 14.2.2
+iosAirshipVersion = 16.0.3
 
 # Urban Airship Android SDK version
-androidAirshipVersion = 14.1.1
+androidAirshipVersion = 16.0.0
```
Comment (oristanovic) [inline airship.properties:9]: should use the latest versions

File: airship.properties [hunk 6]
```diff
@@ -2,10 +2,10 @@
 version = 9.0.0
 
 # Urban Airship iOS SDK version
-iosAirshipVersion = 16.0.3
+iosAirshipVersion = 16.1.0
 
 # Urban Airship Android SDK version
-androidAirshipVersion = 16.0.0
+androidAirshipVersion = 16.1.0
```
Comment (khmMouna) [inline airship.properties:8]: A new version pushed out yesterday (16.1.1)

**Review by khmMouna — APPROVED** [review]

**Pull request merged**